### PR TITLE
Metric type query

### PR DESCRIPTION
**Pull request by micmarty**
Issue #46

- Add optional query for endpoint:
```
allowed string values for metric_type are declared as enum, e.g. mem_util
/nodes/{hostname}/metrics/gpu?metric_type=...
/nodes/{hostname}/metrics/gpu -> all gpu metrics
```
- Add basic reusable responses and parameters

### CODE REVIEW

File: tensorhive/api/APIServer.py [hunk 1]
```diff
@@ -3,6 +3,9 @@
 from tensorhive.config import API_CONFIG
 from tensorhive.database import db_session
 from flask_cors import CORS
+from specsynthase.specbuilder import SpecBuilder
```
Comment (roscisz) [inline tensorhive/api/APIServer.py]: These three imports seem unused. Also, no corresponding dependencies in setup.

File: tensorhive/api/api_specification.yml [hunk 2]
```diff
@@ -148,3 +170,38 @@ definitions:
...
+    required: true
+    type: string
+  gpuMetricTypeQuery:
+    description: Metric type. If not present, queries for all metrics
+    in: query
+    name: metric_type
+    required: false
+    type: string
+    enum:
+      - name
```
Comment (roscisz) [inline tensorhive/api/api_specification.yml]: Maybe these metric names should match the keys in the map returned in the case when no metric is specified, (utilization.gpu [%], power.draw [W]...). Maybe even those keys should be modified, this enum here looks better.

Comment (micmarty) [reply tensorhive/api/api_specification.yml]: Fix in a7839e47b8ede8086dcf107acf328230b98147c4
That's what you had in mind?

File: tensorhive/api/APIServer.py (hunk 1 above)
Comment (micmarty) [reply tensorhive/api/APIServer.py]: Fix in b499874eb9fab4e1fcab80803522b065e733f42b

File: tensorhive/api/api_specification.yml (hunk 2 above)
Comment (roscisz) [reply tensorhive/api/api_specification.yml]: Exactly

**Pull request merged**